src/YieldDisburser.sol

```diff
@@ -128,25 +125,26 @@ contract YieldDisburser is OwnableUpgradeable {
         require(total == 100, "Total must equal 100");
         if (holderToDistribution[holder].length > 0) {
             delete holderToDistribution[holder];
+        } else {
+            breadchainVoters.push(holder);
```

Comment:
tbh I don't love the design of this logic. It's not very intuitive and seems to do a lot of unnecessary operations. Why are we explicitly reseting `holderToDistribution[holder]` only to reassign it at `holderToDistribution[holder] = percentages;`? Why not just reassign it? Is it so we can "push" the voter to `breadchainVoters` the first time a vote is cast by that address? If so that's incredibly non-intuitive. IMO this should be rewritten

Reply:
I see what you're saying, I'll try to refactor this